8.cytopick_analysis/scripts/feature_median_image_find.py

```diff
@@ -0,0 +1,363 @@
+import tifffile as tf
+from cytocherrypick import cherrypick
+from PIL import Image, ImageEnhance
+
+# In[2]:
+
+
+# parameters
+cell_type = "SHSY5Y"
+feature = "Nuclei_Texture_SumVariance_CorrGasdermin_3_01_256"
+
+
+# In[3]:
+
+
+# define directories
+# where the images are
+image_dir_path = pathlib.Path(
+    "/media/lippincm/18T/interstellar_data/70117_20230210MM1_Gasdermin514_CP_BC430856__2023-03-22T15_42_38-Measurement1/2.IC/"
+)
+# if path does not exist, create it
+image_dir_path.mkdir(parents=True, exist_ok=True)
+
+image_out_dir_path = pathlib.Path("../figures/")
+# if path does not exist, create it
+image_out_dir_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[4]:
+
+
+df_path = pathlib.Path(f"../../data/{cell_type}_sc_norm_fs.parquet")
+# read in the data
+df = pd.read_parquet(df_path)
+
+df_no_fs_path = pathlib.Path(f"../../data/{cell_type}_sc.parquet")
+# read in the data
+df_no_fs = pd.read_parquet(df_no_fs_path)
+
+
+# In[5]:
+
+
+df["Nuclei_Location_Center_X"] = df_no_fs["Nuclei_Location_Center_X"]
+df["Nuclei_Location_Center_Y"] = df_no_fs["Nuclei_Location_Center_Y"]
+df["Cytoplasm_AreaShape_BoundingBoxMaximum_X"] = df_no_fs[
+    "Cytoplasm_AreaShape_BoundingBoxMaximum_X"
+]
+df["Cytoplasm_AreaShape_BoundingBoxMaximum_Y"] = df_no_fs[
+    "Cytoplasm_AreaShape_BoundingBoxMaximum_Y"
+]
+df["Cytoplasm_AreaShape_BoundingBoxMinimum_X"] = df_no_fs[
+    "Cytoplasm_AreaShape_BoundingBoxMinimum_X"
+]
+df["Cytoplasm_AreaShape_BoundingBoxMinimum_Y"] = df_no_fs[
+    "Cytoplasm_AreaShape_BoundingBoxMinimum_Y"
+]
+
+
+# In[6]:
+
+
+median, median_index = cherrypick.find_median(df=df, feature_name=feature)
+# get the row of the median
+median_row = df.iloc[median_index]
+image_id = median_row["Metadata_ImageNumber"]
+fov_id = median_row["Metadata_Site"].astype(str)
+cell_id = median_row["Metadata_Cells_Number_Object_Number"]
+well_id = median_row["Metadata_Well"]
+row_id = well_id[0]
+column_id = well_id[1:]
+center_x = median_row["Nuclei_Location_Center_X"]
+center_y = median_row["Nuclei_Location_Center_Y"]
+# median_row['Cytoplasm_']
+# 'Cytoplasm_AreaShape_BoundingBoxMaximum_X', 'Cytoplasm_AreaShape_BoundingBoxMaximum_Y', 'Cytoplasm_AreaShape_BoundingBoxMinimum_X', 'Cytoplasm_AreaShape_BoundingBoxMinimum_Y'
+max_x_box = median_row["Cytoplasm_AreaShape_BoundingBoxMaximum_X"]
+max_y_box = median_row["Cytoplasm_AreaShape_BoundingBoxMaximum_Y"]
+min_x_box = median_row["Cytoplasm_AreaShape_BoundingBoxMinimum_X"]
+min_y_box = median_row["Cytoplasm_AreaShape_BoundingBoxMinimum_Y"]
+print(median_index)
+print(
+    f"Median: {median}",
+    f"Image ID: {image_id}",
+    f"Cell ID: {cell_id}",
+    f"Well ID: {well_id}",
+    f"row_id: {row_id}",
+    f"fov_id: {fov_id}",
+    f"column id: {column_id}",
+    f"Center X: {center_x}",
+    f"Center Y: {center_y}",
+    f"Max X: {max_x_box}",
+    f"Max Y: {max_y_box}",
+    f"Min X: {min_x_box}",
+    f"Min Y: {min_y_box}",
+    sep="\n",
+)
+
+
+# In[7]:
+
+
+well_dict = {
```

Comment:
This is a very interesting way of doing this! I might look into this for myself!